src/wp-includes/kses.php

```diff
@@ -1767,7 +1767,8 @@ function wp_kses_bad_protocol( $content, $allowed_protocols ) {
 	// Short-circuit if the string starts with `https://` or `http://`. Most common cases.
 	if (
 		( str_starts_with( $content, 'https://' ) && in_array( 'https', $allowed_protocols, true ) ) ||
-		( str_starts_with( $content, 'http://' ) && in_array( 'http', $allowed_protocols, true ) )
+		( str_starts_with( $content, 'http://' ) && in_array( 'http', $allowed_protocols, true ) ) ||
+		( str_starts_with( $content, '#:~:text' ) )
```

Comment:
This will make URLs starting with a text fragment impossible to disable. Is there any specific reason for that? If not, thinking the last line should be replaced with:
`( str_starts_with( $content, '#:~:text' ) && in_array( '#:~:text', $allowed_protocols, true ) )`

Then the `#:~:text` string should be added to the default `$allowed_protocols`.